### PR TITLE
PASS1-94: Prevent installing user-signed firmware if no user-key installed

### PR DESCRIPTION
**Pull request by coreylakey**

### CODE REVIEW

File: ports/stm32/boards/Passport/modfoundation.c [hunk 1]
```diff
@@ -1303,7 +1303,10 @@ System_validate_firmware_header(mp_obj_t self, mp_obj_t header)
             vstr_add_strn(&vstr, (const char*)new_fwhdr->info.fwdate, strlen((const char*)new_fwhdr->info.fwdate));
             tuple[2] = mp_obj_new_str_from_vstr(&mp_type_str, &vstr);
 
-            return mp_obj_new_tuple(3, tuple);
+            // Is this user-signed firmware?
+            tuple[3] = (new_fwhdr->signature.pubkey1 == FW_USER_KEY) ? mp_const_true : mp_const_false;
```
Comment (FoundationKen) [inline ports/stm32/boards/Passport/modfoundation.c:1307]: The check on line 1287 should ensure that `pubkey1` is never `FW_USER_KEY`, so we can just return `mp_const_false` here.

File: ports/stm32/boards/Passport/modules/actions.py [hunk 2]
```diff
@@ -371,12 +371,19 @@ def no_pubkeys(filename):
...
             if not is_valid:
                 system.turbo(False)
                 await ux_show_story('Firmware header is invalid.\n\n{}'.format(error_msg), title='Error', left_btn='BACK', right_btn='OK', center=True, center_vertically=True)
                 return
 
+            if is_user_signed:
+                pubkey_result, pubkey = read_user_firmware_pubkey()
+                if not pubkey_result or is_all_zero(pubkey):
+                    system.turbo(False)
+                    await ux_show_story('Install a Developer PubKey before loading non-Foundation firmware.\n\n', title='Error', left_btn='BACK', right_btn='OK', center=True, center_vertically=True)
```
Comment (FoundationKen) [inline ports/stm32/boards/Passport/modules/actions.py:384]: I changed this text, so you'll need to pull before submitting any other changes.

File: ports/stm32/boards/Passport/modfoundation.c (hunk 1 above)
Comment (coreylakey) [reply ports/stm32/boards/Passport/modfoundation.c:1307]: Done :+1:

File: ports/stm32/boards/Passport/modules/actions.py (hunk 2 above)
Comment (coreylakey) [reply ports/stm32/boards/Passport/modules/actions.py:384]: :+1:

**Review by FoundationKen — APPROVED** [review]

**Pull request merged**